salt/modules/debbuild.py

```diff
@@ -416,20 +442,66 @@ def make_repo(repodir, keyid=None, env=None):
     with salt.utils.fopen(repoconfdist, 'w') as fow:
         fow.write('{0}'.format(repocfg_dists))
 
+    local_fingerprint = None
+    local_keyid = None
+
     if keyid is not None:
         with salt.utils.fopen(repoconfdist, 'a') as fow:
             fow.write('SignWith: {0}\n'.format(keyid))
 
+        # gpg keys should have been loaded as part of setup
+        # retrieve specified key, obtain fingerprint and preset passphrase
+        local_keys = __salt__['gpg.list_keys'](user=runas, gnupghome=gnupghome)
+        for gpg_key in local_keys:
+            if keyid == gpg_key['keyid'][8:]:
```

Comment:
Since the `keyid` kwarg is set to None by default, any failure to set that a string will never match here and that case does not appear to be handled below.